### ISSUE TITLE
Produced SurfaceTexture is only 1 color

### ISSUE THREAD

**Issue by Bosa17** [ChillingVan/android-openGL-canvas#96]
I am trying to use sceneform's startMirroringtoSurface method on the GLSurfaceTextureProducerView inherited views following MediaPlayer exmple. But I am getting single color results when using canvas.drawSurfaceTexture

Please help me to understand how this is happening or how I can rectify it

**Comment by Bosa17** [comment]
I think it has to do with wrong UV mapping of the texture

**Comment by Bosa17** [comment]
I think it is only sampling the color of the left bottom pixel of the image and displaying it as whole

**Comment by ChillingVan** [comment]
Hello, which texture do you draw? The producedSurfaceTexture or something defined by yourself? And what's your producedSurfaceTexture?

**Comment by ChillingVan** [comment]
And what about the sample code of the MediaPlayer？Does it work on your device?

**Comment by Bosa17** [comment]
Yes, the media player sample works fine. I follow the same code structure that is shown in the sample. I use the producedSurfaceTexture as tge second arg in the drawSurfaceTexture function

**Comment by ChillingVan** [comment]
Could you share your code as a project or snippet with enough information? 
In the media player example, the surface texture is provided to the media surface. So do you create your surface? 
And does sceneForm do something to the surface?  Maybe not only the opengl canvas manipulate the surface.

**Comment by Bosa17** [comment]
https://github.com/Bosa17/TimBerr

The concerned activities are ArtWorkDisplayActivity and PreviewSurfaceTextureView

**Comment by Bosa17** [comment]
Sceneform just renders output to the surface.

**Comment by ChillingVan** [comment]
Do you know which color format outputed from the sceneform? Usually for media or camera the surface is YUV format so the ProducedTextureTarget should be GLES11Ext.GL_TEXTURE_EXTERNAL_OES. But for format like Bitmap ARGB8888, the ProducedTextureTarget  should be GLES20.GL_TEXTURE_2D.
You can set the ProducedTextureTarget  when the GLSurfaceTextureProducerView creates.

**Comment by Bosa17** [comment]
I tried using both. The GLES20.GL_TEXTURE_2D produces no result and onGlDraw is only called once. The other format produces the same single color results

**Comment by Bosa17** [comment]
Is there any scenario where RawTexture just gives 1 cokor resukt?

**Comment by ChillingVan** [comment]
> I tried using both. The GLES20.GL_TEXTURE_2D produces no result and onGlDraw is only called once. The other format produces the same single color results

onGlDraw only called once. That's to say, onFrameAvailable is only called once. Maybe you can check whether the sceneForm has triggered rendering the surface

**Comment by ChillingVan** [comment]
> Is there any scenario where RawTexture just gives 1 cokor resukt?

It may not has relation with the RawTexture..

**Comment by ChillingVan** [comment]
I check the startMirroringToSurface again.
And I think there should be one other thing in the whole part. That is the media or the Camera. 
For current code, there is nothing for the input surface. So neither the sceneForm nor the PreviewSurfaceTextureView has input for the surface so they cannot render anything from the surface
